api/src/main/java/org/openmrs/module/htmlformentryui/HtmlFormUtil.java

```diff
@@ -55,6 +69,19 @@ public static HtmlForm getHtmlFormFromUiResource(ResourceFactory resourceFactory
         return getHtmlFormFromResourceXml(formService, htmlFormEntryService, xml);
     }
 
+
+    // the new method above with "encounter" is preferred if an encounter is available, see: https://issues.openmrs.org/browse/HTML-768
+    public static HtmlForm getHtmlFormFromUiResource(ResourceFactory resourceFactory, FormService formService, HtmlFormEntryService htmlFormEntryService, String providerAndPath) throws IOException {
+        // using new Encounter() is a bit of a hack, used for a proper method signature hack; safe because the called method
+        // perform a null check on encounter.getForm()
+        return getHtmlFormFromUiResource(resourceFactory, formService, htmlFormEntryService, providerAndPath, new Encounter());
```

Comment:
Don't love the "new Encounter()".  Could you achieve the same effect by doing "(Encounter)null" ?

Reply:
Oh, cool, I never realized you could do that... it appears to work, thanks!